core/domain/question_domain.py

```diff
@@ -296,3 +299,67 @@ def to_dict(self):
             'question_id': self.question_id,
             'skill_id': self.skill_id,
         }
+
+
+class QuestionCommitLogEntry(object):
+    """Value object representing a commit to an question."""
+
+    def __init__(
+            self, created_on, last_updated, user_id, username, question_id,
+            commit_type, commit_message, commit_cmds, version,
+            post_commit_status, post_commit_is_private):
+        """Initializes a QuestionCommitLogEntry domain object.
+
+        Args:
+            created_on: datetime.datetime. Date and time when the question
+                commit was created.
+            last_updated: datetime.datetime. Date and time when the question
+                commit was last updated.
+            user_id: str. User id of the user who has made the commit.
+            username: str. Username of the user who has made the commit.
+            question_id: str. Id of the question.
+            commit_type: str. The type of commit.
+            commit_message: str. A description of changes made to the
+                question.
+            commit_cmds: list(dict). A list of commands, describing changes
+                made in this model, which should give sufficient information to
+                reconstruct the commit. Each dict always contains the following
+                key:
+                    - cmd: str. Unique command.
+                and then additional arguments for that command.
+            version: int. The version of the question after the commit.
+            post_commit_status: str. The new question status after the
+                commit.
+            post_commit_is_private: bool. Whether the question is private
+                after the edit event.
+        """
+        self.created_on = created_on
+        self.last_updated = last_updated
+        self.user_id = user_id
+        self.username = username
+        self.question_id = question_id
+        self.commit_type = commit_type
+        self.commit_message = commit_message
+        self.commit_cmds = commit_cmds
+        self.version = version
+        self.post_commit_status = post_commit_status
+        self.post_commit_is_private = post_commit_is_private
+
+    def to_dict(self):
```

Comment:
Is this needed, are we using this anywhere?

Reply:
As of now we aren't.